examples/tutorials/root_of_trust/questionable_service_milestone_one/main.c

```diff
@@ -0,0 +1,117 @@
+// Hardware Root of Trust (HWRoT) Hardfault Driver Application
+//
+// When selected by the main screen HWRoT Demo application, calls a command
+// on a driver which is intended to hardfault all applications.
+
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <libtock/kernel/ipc.h>
+#include <libtock/tock.h>
+
+#define LOG_WIDTH      32
+
+bool started          = false;
+bool log_done         = false;
+size_t screen_service = -1;
+char log_buf[LOG_WIDTH] __attribute__((aligned(LOG_WIDTH)));
+
+const char SCREEN_SERVICE_NAME[] = "org.tockos.tutorials.attestation.screen";
+
+static void ipc_callback(__attribute__ ((unused)) int   pid,
+                         __attribute__ ((unused)) int   len,
+                         __attribute__ ((unused)) int   arg2,
+                         __attribute__ ((unused)) void* ud) {
+  started = true;
+}
+
+static void log_done_callback(__attribute__ ((unused)) int   pid,
+                              __attribute__ ((unused)) int   len,
+                              __attribute__ ((unused)) int   arg2,
+                              __attribute__ ((unused)) void* ud) {
+  log_done = true;
+}
+
+static void wait_for_start(void) {
+  // Register an IPC callback and wait for it to be called by the
+  // screen app based on the user's app selection.
+  ipc_register_service_callback("org.tockos.tutorials.attestation.questionable", ipc_callback,
+                                NULL);
+  yield_for(&started);
+}
+
+static int setup_logging() {
+  returncode_t ret;
+
+  // Find the PID of the screen logging service
+  ret = ipc_discover(SCREEN_SERVICE_NAME, &screen_service);
+  if (ret != RETURNCODE_SUCCESS) {
+    printf("Screen logging service not found.\n");
+    return ret;
+  }
+
+  // Set up a callback and share so we can supply log messages
+  // and know when they've been completely logged.
+  ipc_register_client_callback(screen_service, log_done_callback, NULL);
+  ipc_share(screen_service, log_buf, LOG_WIDTH);
+
+  return 0;
+}
+
+static int log_to_screen(const char* message) {
+  returncode_t ret;
+
+  // Copy up to the log buffer's size of the message, with room for a null byte.
+  uint16_t len = strnlen(message, sizeof(log_buf) - 1);
+  memcpy(log_buf, message, len);
+
+  // Add the null byte.
+  log_buf[len] = '\0';
```

Comment:
```suggestion
  strlcpy(log_bug, message, LOG_WIDTH);
```

I think this is a simpler version of the same?

Reply:
Good catch (and TIL!), committing from GH then will test locally to triple check (will report here)